pydantic_ai_slim/pydantic_ai/agent.py

```diff
@@ -334,6 +340,14 @@ def __init__(
         self._default_retries = retries
         self._max_result_retries = output_retries if output_retries is not None else retries
         self._mcp_servers = mcp_servers
+        self._builtin_tools: list[AbstractBuiltinTool] = []
+
+        for tool in builtin_tools:
+            if tool == 'web-search':
+                self._builtin_tools.append(WebSearchTool())
+            else:
+                self._builtin_tools.append(tool)
```

Comment:
It's easier to not have to handle string on the models, so we already do the transformation here.